esp-hal/MIGRATING-0.23.md

```diff
@@ -277,3 +278,17 @@ NOTE: Async support is only supported in ESP32C3 and ESP32C6 for now
 - Adc<'d, ADC>
 + Adc<'d, ADC, Blocking>
 ```
+
+## time API changes
+
+ESP-HAL no longer publicly exposes `fugit` and no longer exposes the concept of a `tick`.
+This comes with a number of changes:
+
+- The `RateExtU32` and similar traits are no longer used, which means `.kHz()` and similar suffix
+  conversions are no longer available. A number of matching constructors are available. For example,
+  instead of `1.MHz()` you need to write `Rate::from_mhz(1)`.
```

Comment:
So as per the parent issue, let's bikeshed this a bit.

Imo this still looks the cleanest, and any confusion can be cleared up quite easily with a doc comment. However, I understand the want for correctness here. I want to avoid capitalization if possible, the rest of our APIs are lowercase.

Of the top of my head suggestions:

- `mghz`
- `mega_hz`
- `mega_hertz`

Other suggestions are most welcome!

Reply:
+1 for `mhz`

Reply:
`mghz` is just confusing, `mega_hz` or `mega_hertz` look sloppy as I think the prefixes are supposed to be written together with the unit, like `megahertz`. We don't spell out `decibels` either in the ADC driver, although this current PR will set the precedent I guess.

`mhz` is good enough as it's consistent with the Rust [guidelines](https://rust-lang.github.io/api-guidelines/naming.html#c-case). To quote:

> In snake_case, acronyms and contractions are lower-cased: is_xid_start.

Also to quote:

> This guideline should be easy!

Reply:
I think the guideline suggestion has resolved the concerns here for me, I suppose I should have looked there first :D.